src/LDPCDecoders.jl

```diff
@@ -10,26 +10,23 @@ using RowEchelon
 
 
 # Write your package code here.
-export parity_to_generator
+export
+    parity_to_generator, hamming_to_parity, repetition_to_parity, 
+    bp_decode, bp_simulate, it_decode, parity_check_matrix, save_pcm, 
+    load_pcm, syndrome_decode, syndrome_simulate, syndrome_it_decode, 
+    syndrome_it_simulate, syndrome_decode!, decode!, BeliefPropagationSetup, 
+    BeliefPropagationDecoder, BeliefPropagationDecoderSparse, batchdecode!
```

Comment:
```suggestion
export
    batchdecode!, decode!,
    BeliefPropagationDecoder 
```

Let's be as sparing as possible, that way we do not shoot ourselves in the foot with something we do not like but need to support long term.

Reply:
I also think we should keep `BeliefPropagationSetup` private. The user should not know about this, it is just some internal "scratch" space used by the algorithm.

Reply:
aaaaand `BeliefPropagationDecoderSparse` is gone now too